addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -641,6 +641,14 @@ export function getSelectedNodes(editable) {
  */
 export function getDeepRange(editable, { range, sel, splitText, select, correctTripleClick } = {}) {
     sel = sel || editable.ownerDocument.getSelection();
+    // if browser wrongly return br element as anchor node, replace by its parent
+    if (sel.type === "Caret" && sel.anchorNode.nodeName === "BR") {
+        const newRange = document.createRange();
+        newRange.selectNodeContents(sel.anchorNode.parentElement);
+        sel.removeAllRanges();
+        sel.addRange(newRange);
+        sel.collapseToStart();
+    }
```

Comment:
I don't think this is the right place for this fix because I don't think it is the responsibility of `getDeepRange` to _fix_ the range if it is wrong to begin with. However, I don't think we have another function that could do it so... maybe it makes sense if you consider that the given range is in fact _too_ deep, in which case it could fall into getDeepRange's territory to make it back to "as deep as possible", therefore making it _less_ deep in this case, if it makes sense.

That being said, we usually use `isCollapsed` rather than checking the `type`. Also, we do have an util to set the selection so I suggest to use it rather than making multiple calls to the native api.

Reply:
Thanks for checking! I will adjust it with the utils :pray:

Reply:
You made me doubt so I double checked this. It turns out the document used in `setSelection` is not the global `document` but the document owning the node that you are selecting. So I checked the `sel` parameter in `getDeepRange` and it turns out it is kind of weird, because it is not supported to call `getDeepRange` with an `editable` that is not in the same document that the provided selection. In a way, the parameter is useless because the selection could always be accessed from the provided `editable`, but in truth the same issue exists with the `range` parameter and it is useful to give another range than the current one, so removing the parameter won't help in that case. Long story short: It's fine. It's going to work in all cases for which `getDeepRange` itself is supposed to work. Thanks for being careful ! ❤️